d2renderers/textmeasure/textmeasure.go

```diff
@@ -97,49 +99,57 @@ type Ruler struct {
 //   })
 //   txt := text.New(orig, text.NewAtlas(face, text.ASCII))
 func NewRuler() (*Ruler, error) {
-	lineHeights := make(map[d2fonts.Font]float64)
-	tabWidths := make(map[d2fonts.Font]float64)
-	atlases := make(map[d2fonts.Font]*atlas)
+	origin := geo.NewPoint(0, 0)
+	r := &Ruler{
+		Orig:             origin,
+		Dot:              origin.Copy(),
+		LineHeightFactor: 1.,
+		lineHeights:      make(map[d2fonts.Font]float64),
+		tabWidths:        make(map[d2fonts.Font]float64),
+		atlases:          make(map[d2fonts.Font]*atlas),
+		ttfs:             make(map[d2fonts.Font]*truetype.Font),
+	}
 
 	for _, fontFamily := range d2fonts.FontFamilies {
-		for _, fontSize := range d2fonts.FontSizes {
-			for _, fontStyle := range d2fonts.FontStyles {
-				font := d2fonts.Font{
-					Family: fontFamily,
-					Style:  fontStyle,
-				}
-				if _, ok := d2fonts.FontFaces[font]; !ok {
-					continue
-				}
+		for _, fontStyle := range d2fonts.FontStyles {
+			font := d2fonts.Font{
+				Family: fontFamily,
+				Style:  fontStyle,
+			}
+			// Note: FontFaces lookup is size-agnostic
+			if _, ok := d2fonts.FontFaces[font]; !ok {
+				continue
+			}
+			if _, loaded := r.ttfs[font]; !loaded {
 				ttf, err := truetype.Parse(d2fonts.FontFaces[font])
 				if err != nil {
 					return nil, err
 				}
-				// Added after, since FontFaces lookup is size-agnostic
-				font.Size = fontSize
-				face := truetype.NewFace(ttf, &truetype.Options{
-					Size: float64(fontSize),
-				})
-				atlas := NewAtlas(face, ASCII)
-				atlases[font] = atlas
-				lineHeights[font] = atlas.lineHeight
-				tabWidths[font] = atlas.glyph(' ').advance * TAB_SIZE
+				r.ttfs[font] = ttf
+			}
+
+			for _, fontSize := range d2fonts.FontSizes {
```

Comment:
do we still need to preload this? or would it now just load dynamically?

Reply:
updated so we can just load everything dynamically